src/shogun/lib/type_case.h

```diff
@@ -75,10 +75,49 @@ namespace shogun
 {
 	typedef std::unordered_map<std::type_index, TYPE> typemap;
 
+	namespace type_internal
+	{
+		std::string demangled_type(const char* name)
```

Comment:
wait, didn't we have a function like this somewhere already?

Reply:
yes, it is similar, but it is a templated function that requires the type. Here we don't have the type, but have the `std::type_info::name` instead (and we cant infer type from that), so I needed this custom function that I hid in this namespace.

Reply:
any way to re-use some redundant code?

Reply:
especially the guard/macro bit....and only if it doesnt cause more trouble than it solves

Reply:
I will give it a go! I can add both implementations in Any.h

Reply:
I put it in Any.cpp as I was getting a duplicate symbol linker error.